dot/parachain/prospective-parachains/fragment_chain.go

```diff
@@ -39,6 +39,36 @@ type candidateEntry struct {
 	state              candidateState
 }
 
+var _ HypoteticalOrConcrete = (*candidateEntry)(nil)
```

Comment:
Why do we need this interfcae? when only 1 struct is implementing the interface. Can't we use candidateEntry directly?

i know that in Rust they have created interface but do we really need this?